### PR TITLE
Add Resource pack prompt message option (1.17+)

### PR DESCRIPTION
**Pull request by LemonCaramel**
This PR can set the resource pack prompt message.
It will be useful for servers that require resource packs with the "force-use-1_17-resource-pack" option.

I've tried to write a neat code, but I don't think it's neat.
Feedback on this PR is welcome.

image

### CODE REVIEW

File: common/src/main/java/com/viaversion/viaversion/util/Config.java [hunk 1]
```diff
@@ -207,4 +212,15 @@ public double getDouble(String key, double def) {
             return new ArrayList<>();
         }
     }
+
+    public @Nullable JsonElement getSerializedComponent(String key) {
+        final Object o = this.config.get(key);
+        if (o != null && !((String) o).isEmpty()) {
+            return GSON_SERIALIZER.serializeToTree(Component.text((String) o)
```
Comment (KennyTV) [inline common/src/main/java/com/viaversion/viaversion/util/Config.java:219]: Use the legacy section deserializer and directly serialize again without the rest

File: api/src/main/java/com/viaversion/viaversion/api/configuration/ViaVersionConfig.java [hunk 2]
```diff
@@ -414,4 +415,11 @@
      * @return true if enabled
      */
     boolean isForcedUse1_17ResourcePack();
+
+    /**
+     *  Get the message that is sent when a user displays a resource pack prompt.
+     *
+     * @return cached serialized component
+     */
+    JsonElement getResourcePack1_17PromptMsg();
```
Comment (KennyTV) [inline api/src/main/java/com/viaversion/viaversion/api/configuration/ViaVersionConfig.java:424]:
```suggestion
    JsonElement get1_17ResourcePackPrompt();
```

File: common/src/main/java/com/viaversion/viaversion/configuration/AbstractViaConfig.java [hunk 3]
```diff
@@ -136,6 +138,7 @@ protected void loadFields() {
         instantRespawn = getBoolean("use-1_15-instant-respawn", false);
         ignoreLongChannelNames = getBoolean("ignore-long-1_16-channel-names", true);
         forcedUse1_17ResourcePack = getBoolean("forced-use-1_17-resource-pack", false);
+        resourcePack1_17PromptMessage = getSerializedComponent("resource-pack-1_17-prompt-message");
```
Comment (KennyTV) [inline common/src/main/java/com/viaversion/viaversion/configuration/AbstractViaConfig.java:141]:
```suggestion
        resourcePack1_17PromptMessage = getSerializedComponent("resource-pack-1_17-prompt");
```

**Review by KennyTV — CHANGES_REQUESTED** [review]

File: common/src/main/java/com/viaversion/viaversion/util/Config.java (hunk 1 above)
Comment (LemonCaramel) [reply common/src/main/java/com/viaversion/viaversion/util/Config.java:219]: May I know what the "legacy section deserializer" means? [this?](https://github.com/ViaVersion/ViaVersion/blob/master/common/src/main/java/com/viaversion/viaversion/protocols/protocol1_13to1_12_2/ChatRewriter.java#L32)

Comment (KennyTV) [reply common/src/main/java/com/viaversion/viaversion/util/Config.java:219]: `LegacyComponentSerializer.legacySection()`

Comment (tomcraft) [reply common/src/main/java/com/viaversion/viaversion/util/Config.java:219]: it's Adventure serializer for legacy strings "§cHello §bWorld"  : LegacyComponentSerializer. legacySection()

Comment (LemonCaramel) [reply common/src/main/java/com/viaversion/viaversion/util/Config.java]: Thank you! I just pushed the commit.

**Comment by KennyTV** [comment]
The actual use of the method and the default config entry are gone? 👀

**Comment by LemonCaramel** [comment]
OMG... I haven't been able to sleep well lately. so I think I made a mistake. I'll fix it in a minute.

**Review by KennyTV — APPROVED** [review]
Thank you

**Pull request merged**